rpc/ethereum/backend/backend.go

```diff
@@ -696,16 +683,17 @@ func (e *EVMBackend) GetTransactionByHash(txHash common.Hash) (*types.RPCTransac
 		return nil, errors.New("invalid ethereum tx")
 	}
 
+	msgIndex, attrs := types.FindTxAttributes(res.TxResult.Events, hexTx)
+	if msgIndex < 0 {
+		return nil, fmt.Errorf("ethereum tx not found in msgs: %s", hexTx)
+	}
+
 	tx, err := e.clientCtx.TxConfig.TxDecoder()(res.Tx)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(tx.GetMsgs()) != 1 {
-		return nil, errors.New("invalid ethereum tx")
-	}
-
-	msg, ok := tx.GetMsgs()[0].(*evmtypes.MsgEthereumTx)
+	msg, ok := tx.GetMsgs()[msgIndex].(*evmtypes.MsgEthereumTx)
```

Comment:
check out of bounds?

Reply:
I think it should be fine since the `msgIndex` is inferred from tx events, added a comment about it.